### PR TITLE
samples: boards: nrfx: Align code with the recent nrfx_gpiote API

### LINKED ISSUE #46645: NRFX samples use deprecated API
Zephyr `3.0.0` uses the NRFX module in version `2.7.0`. In that version `nrfx_gpiote_in_init()` got marked as deprecated, see https://github.com/NordicSemiconductor/nrfx/blob/v2.7.0/drivers/include/nrfx_gpiote.h#L634.
How ever the samples still use the deprectaed API instead of showing how to use the new API, see https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/nrf/nrfx/src/main.c#L59

### PR DESCRIPTION
**Pull request by anangl**
Replace calls to nrfx_gpiote functions deprecated in nrfx 2.6.0
with recommended equivalents.
On the occasion, switch to using available `nrfx_gppi_*` functions
instead of separate paths for DPPI and PPI channels.

Fixes #46645

### CODE REVIEW

File: samples/boards/nrf/nrfx/src/main.c [hunk 1]
```diff
@@ -41,76 +44,101 @@ void main(void)
...
 	 */
-	nrfx_gppi_channel_endpoints_setup(channel,
+	nrfx_gppi_channel_endpoints_setup(ppi_channel,
 		nrfx_gpiote_in_event_addr_get(INPUT_PIN),
 		nrfx_gpiote_out_task_addr_get(OUTPUT_PIN));
 
 	/* Enable (D)PPI channel. */
 #if defined(DPPI_PRESENT)
-	err = nrfx_dppi_channel_enable(channel);
+	err = nrfx_dppi_channel_enable(ppi_channel);
```
Comment (nika-nordic) [inline samples/boards/nrf/nrfx/src/main.c:136]: can't we switch to `nrfx_gppi_channels_enable` on the occasion? Or this is out of scope of this PR?

**Review by nika-nordic — DISMISSED** [review]

File: samples/boards/nrf/nrfx/src/main.c (hunk 1 above)
Comment (anangl) [reply samples/boards/nrf/nrfx/src/main.c:136]: Good point. I'll change this.

File: samples/boards/nrf/nrfx/src/main.c [hunk 2]
```diff
@@ -41,78 +44,95 @@ void main(void)
...
 
 	LOG_INF("nrfx_gpiote initialized");
 
 	/* Allocate a (D)PPI channel. */
 #if defined(DPPI_PRESENT)
-	uint8_t channel;
-	err = nrfx_dppi_channel_alloc(&channel);
+	uint8_t ppi_channel;
+
+	err = nrfx_dppi_channel_alloc(&ppi_channel);
```
Comment (nika-nordic) [inline samples/boards/nrf/nrfx/src/main.c:115]: sorry for not spotting this earlier, but we have `nrfx_gppi_channel_alloc` as well

Comment (anangl) [reply samples/boards/nrf/nrfx/src/main.c]: Applied.

**Review by nika-nordic — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
